internal/patroni/config.go

```diff
@@ -552,7 +552,10 @@ func instanceYAML(
 
 	if !ClusterBootstrapped(cluster) {
 		// if restore status exists, then a restore occurred an the "existing" method is used
-		if cluster.Status.PGBackRest != nil && cluster.Status.PGBackRest.Restore != nil {
+		if (cluster.Status.PGBackRest != nil && cluster.Status.PGBackRest.Restore != nil) ||
+			(cluster.Spec.DataSource != nil && cluster.Spec.DataSource.ExistingVolumes != nil &&
+				cluster.Spec.DataSource.ExistingVolumes.ExistingPGDataVolume != nil &&
+				cluster.Spec.DataSource.ExistingVolumes.ExistingPGDataVolume.Directory != "") {
```

Comment:
I agree it's a lot, but I don't know about moving it. I believe that function would only be used here, and I'm afraid further abstracting it would make the code more difficult to parse.

Reply:
Sure, I wish there was a better way to make this check. Its annoying to have to check each pointer on the way down to the piece you care about. :thought_balloon: